node_modules/oae-esup/loodle/js/loodle.js

```diff
@@ -0,0 +1,742 @@
+                if (key === type) {
+                    if (isAlreadySend) {
+                        alreadySend = true;
+                    } else {
+                        notifications[key] = true;
+                        notificationType = key;
+                    }
+                }
+            });
+
+            if (!alreadySend) {
+                oae.api.loodle.sendNotifications(widgetData.id, notificationType, function (err) {
+                    if (err) {
+                        console.error('An error occured : ', err);
+                        oae.api.util.notification(
+                            oae.api.i18n.translate('__MSG__LOODLE_NOTIFICATIONS_NOT_SENT__', 'loodle'),
+                            oae.api.i18n.translate('__MSG__LOODLE_NOTIFICATIONS_COULD_NOT_BE_SENT__', 'loodle'));
+                    }
+                });
+            }
+
+        }
+
+    };
+
+    /**
+     * Update the view of the loodle with the current loodle data
+     *
+     * @param  {Object}     widgetData      Widget data
+     * @param  {String}     locale          Language currently use by the user
+     */
+    var updateLoodleDisplay = function (widgetData, locale) {
+
+        oae.api.loodle.getData(widgetData.loodleData.id, function (err, loodle) {
+
+            if (err) {
+                console.error('An error occured trying to get the loodle data : ', err);
+                oae.api.util.notification(
+                    oae.api.i18n.translate('__MSG__UNEXPECTED_ERROR__', 'loodle'),
+                    oae.api.i18n.translate('__MSG__LOODLE_DISPLAY_COULD_NOT_BE_UPDATED__', 'loodle'));
+                return;
+            }
+
+            // Sort the users array so that the current user always has his/her name at the bottom of the table
+            var index = 0;
+            var currentUser = {};
+            for (var i = 0; i < loodle.users.length; i++) {
+                if (loodle.users[i].id == widgetData.loodleData.currentUser.id) {
+                    index = i;
+                    currentUser = loodle.users[i];
+                }
+            }
+
+
+            loodle.users.splice(index, 1);
+            loodle.users.push(currentUser);
+
+            // We keep a trace of the current user
+            loodle.currentUser = widgetData.loodleData.currentUser;
+
+            // Format loodle data to easily display
+            formatSchedules(loodle, oae.data.me.locale);
+            formatUsers(loodle);
+
+            // Update the loodle view with the updated data
+            updateLoodleView(loodle);
+
+            // Change the active page
+            toggleActiveView('loodle');
+
+            // Update the delete schedule view with the new schedule added
+            updateDeleteScheduleView(loodle);
+
+        });
+
+    };
+
+    /**
+     * Render the loodle updated in the page
+     *
+     * @param  {Object} loodle  Loodle data updated
+     */
+    var updateLoodleView = function (loodle) {
```

Comment:
While this does seem to work, I would really prefer not building html in JS... Can we use a template instead?